var/spack/repos/builtin/packages/conduit/package.py

```diff
@@ -0,0 +1,288 @@
+            host_cfg_fname = self.create_host_config(spec, prefix)
+            cmake_args = []
+            # if we have a static build, we need to avoid
+            # any of spack's default cmake settings related to
+            # rpath
+            if "+shared" in spec:
+                cmake_args.extend(std_cmake_args)
+            else:
+                for arg in std_cmake_args:
+                    if arg.count("RPATH") == 0:
+                        cmake_args.append(arg)
+            cmake_args.extend(["-C", host_cfg_fname, "../src"])
+            cmake(*cmake_args)
+            make()
+            make("install")
+
+    def create_host_config(self, spec, prefix):
+        """
+        This method creates a 'host-config' file that specifies
+        all of the options used to configure and build conduit.
+
+        For more details see about 'host-config' files see:
+            http://software.llnl.gov/conduit/building.html
+        """
+
+        #######################
+        # Compiler Info
+        #######################
+        c_compiler = env["SPACK_CC"]
+        cpp_compiler = env["SPACK_CXX"]
+        f_compiler = None
+
+        if self.compiler.fc:
+            # even if this is set, it may not exist do one more sanity check
+            if os.path.isfile(env["SPACK_FC"]):
+                f_compiler = env["SPACK_FC"]
+
+        #######################################################################
+        # By directly fetching the names of the actual compilers we appear
+        # to doing something evil here, but this is necessary to create a
+        # 'host config' file that works outside of the spack install env.
+        #######################################################################
+
+        sys_type = spec.architecture
+        # if on llnl systems, we can use the SYS_TYPE
+        if "SYS_TYPE" in env:
+            sys_type = env["SYS_TYPE"]
+
+        ##############################################
+        # Find and record what CMake is used
+        ##############################################
+
+        if "+cmake" in spec:
+            cmake_exe = join_path(spec['cmake'].prefix.bin, "cmake")
+        else:
+            cmake_exe = which("cmake")
+            if cmake_exe is None:
+                msg = 'failed to find CMake (and cmake variant is off)'
+                raise RuntimeError(msg)
+            cmake_exe = cmake_exe.command
+
+        host_cfg_fname = "%s-%s-%s.cmake" % (socket.gethostname(),
+                                             sys_type,
+                                             spec.compiler)
+
+        cfg = open(host_cfg_fname, "w")
+        cfg.write("##################################\n")
+        cfg.write("# spack generated host-config\n")
+        cfg.write("##################################\n")
+        cfg.write("# %s-%s\n" % (sys_type, spec.compiler))
+        cfg.write("##################################\n\n")
+
```

Comment:
This whole section could be made a LOT more clear using str.format(*args, **kwargs) available in Python 2.6.  For example:
```
cfg.write("""#######################
# spack generated host-config
#######################
# {sys_type}-{compiler}
##################
""".format(sys_type=sys_type, compiler=spec.compiler)
```
The `cmake_cache_entry()` helper function can also be eliminated this way.